subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/internal/DefaultSwiftXCTestSuite.java

```diff
@@ -36,6 +37,9 @@
     public DefaultSwiftXCTestSuite(String name, ObjectFactory objectFactory, FileOperations fileOperations, ConfigurationContainer configurations, ProjectLayout projectLayout) {
         super(name, fileOperations, objectFactory, configurations);
 
+        Configuration testImplementation = configurations.maybeCreate("testImplementation");
+        testImplementation.extendsFrom(configurations.maybeCreate("implementation"));
```

Comment:
Or if you can get an instance of both component (test and main) you can use `getImplementationDependencies()` that is inherited from `SwiftComponent`.

Reply:
I've updated this to use the `SwiftComponent` objects and moved the logic into the `XCTestConventionPlugin`'s `configureTestedSwiftComponent` method.

One thing I'd like to understand for the future is *why* that's the better way to do it?  It seems like both @big-guy and @lacasseio had an understanding about the responsibilities of these two classes that made them realize that one was better than the other. But, why?

Reply:
Daniel had the better suggestion.

1. It was a coincidence that the configuration for the component-under-test was called "implementation".  It could be called other things (likely if we allow someone to configure the component-under-test) or we might rename it (unlikely).  
1. It was also a coincidence that the test's configuration was called 'testImplementation'.  This name is chosen based on the name of the test suite in `DefaultSwiftComponent`.
1. There isn't a guarantee that there will be a component-under-test defined in the project (which would define the implementation configuration).  `maybeCreate` gets around that by creating the implementation configuration every time, which means you'll get an implementation configuration showing up in `gradle dependencies`, build scans, etc without applying either Swift plugin.

Those were the main things I thought of.